core/src/main/java/org/elasticsearch/search/aggregations/bucket/histogram/DateHistogramParser.java

```diff
@@ -52,6 +51,8 @@ protected DateHistogramAggregationBuilder createFactory(String aggregationName,
             factory.interval((Long) interval);
         } else if (interval instanceof DateHistogramInterval) {
             factory.dateHistogramInterval((DateHistogramInterval) interval);
+        } else {
+            throw new AssertionError();
```

Comment:
I think if you throw a subclass of RuntimeException here it'll get bubbled back to the user and we'll get a bug report if it shows up. AssertionError's problem is that catching it and forwarding it around properly gets us dangerously close to catching OOMs. We probably could make an exception for catching AssertionError but I'm not sure it is worth it when we can just throw a RuntimeException of some sort instead.

Also it just feels weird to throw an assertion error from outside of assertions! Its icky 😨